src/main/scala/devices/i2c/I2CPeriphery.scala

```diff
@@ -19,15 +19,21 @@ trait HasPeripheryI2C extends HasTopLevelNetworks {
   }
 }
 
-trait HasPeripheryI2CBundle extends HasTopLevelNetworksBundle{
-  val outer: HasPeripheryI2C
-  val i2cs = Vec(outer.i2cParams.size, new I2CPort)
+trait HasPeripheryI2CBundle {
+  val i2cs: Vec[I2CPort]
+
+  def toGPIOPins(dummy: Int = 1): Seq[I2CGPIOPort] = i2cs.map { i =>
```

Comment:
Ah, i see.  Well any of the "GPIOPort" which has inputs should allow a synchronizer depth parameter. I think right now only SPIGPIOPort has that.